pygfx/helpers/_gizmo.py

```diff
@@ -0,0 +1,747 @@
+        for ob in [line_y, translate_y, scale_y, arc_zx, translate_zx, rotate_zx]:
+            ob.rotation.set_from_axis_angle(Vector3(0, 0, 1), np.pi / 2)
+        for ob in [line_z, translate_z, scale_z, arc_xy, translate_xy, rotate_xy]:
+            ob.rotation.set_from_axis_angle(Vector3(0, -1, 0), np.pi / 2)
+
+        # Store the objectss
+        self._center_sphere = scale_uniform
+        self._line_children = line_x, line_y, line_z
+        self._arc_children = arc_yz, arc_zx, arc_xy
+        self._translate1_children = translate_x, translate_y, translate_z
+        self._translate2_children = translate_yz, translate_zx, translate_xy
+        self._translate_children = self._translate1_children + self._translate2_children
+        self._scale_children = scale_x, scale_y, scale_z
+        self._rotate_children = rotate_yz, rotate_zx, rotate_xy
+
+        # Lines and arcs are never apaque. That way they're also not
+        # pickable, so they won't be "in the way".
+        for ob in self._line_children + self._arc_children:
+            ob.material.opacity = 0.6
+
+        # Assign dimensions
+        scale_uniform.dim = None
+        for triplet in [
+            self._line_children,
+            self._translate1_children,
+            self._scale_children,
+            self._rotate_children,
+        ]:
+            for i, ob in enumerate(triplet):
+                ob.dim = i
+        for i, ob in enumerate(self._translate2_children):
+            dim = [0, 1, 2]
+            dim.pop(i)
+            ob.dim = tuple(dim)
+
+        # Attach to the gizmo object
+        self.add(*self._line_children)
+        self.add(*self._arc_children)
+        self.add(self._center_sphere)
+        self.add(*self._translate_children)
+        self.add(*self._scale_children)
+        self.add(*self._rotate_children)
+
+    def _on_mode_switch(self):
+        """When the mode is changed, some adjustments are made."""
+
+        # Note: the visibility of certain elements is handled in _update_visibility()
+
+        # Update position of rotation handles
+        rotate_yz, rotate_zx, rotate_xy = self._rotate_children
+        halfway = self._halfway
+        on_arc = halfway * 2**0.5 / 2
+        if self._mode == "screen":
+            rotate_yz.position.set(0, on_arc, 0)
+            rotate_zx.position.set(on_arc, 0, 0)
+            rotate_xy.position.set(on_arc, on_arc, 0)
+        else:
+            rotate_yz.position.set(0, on_arc, on_arc)
+            rotate_zx.position.set(on_arc, 0, on_arc)
+            rotate_xy.position.set(on_arc, on_arc, 0)
+
+    def _highlight(self, object=None):
+        """Change the appearance during interaction for visual feedback
+        on what is being manipulated.
+        """
+        # Reset thickness of all lines
+        for ob in self._line_children + self._arc_children:
+            if ob.material.thickness != 3:
+                ob.material.thickness = 3
```

Comment:
might want to make this a constant as well?

Reply:
Done.